identity/src/main/java/com/android/identity/document/Document.kt

```diff
@@ -223,34 +229,36 @@ class Document private constructor(
     /**
      * Creates a new credential.
      *
-     * This returns an [Credential] which should be sent to the document
+     * This returns an [MdocCredential] which should be sent to the document
      * issuer for certification. Use [Credential.certify] when certification
      * has been obtained.
      *
      * For a higher-level way of managing credentials, see
-     * [DocumentUtil.managedCredentialHelper].
+     * [DocumentUtil.managedSecureAreaBoundCredentialHelper].
      *
      * @param domain a string used to group credentials together.
      * @param secureArea the secure area to use for the credential.
      * @param createKeySettings settings for the credential.
      * @param asReplacementFor if not `null`, replace the given credential
      * with this one, once it has been certified.
-     * @return an [Credential].
+     * @return a [SecureAreaBoundCredential].
      * @throws IllegalArgumentException if `asReplacementFor` is not null and the given
      * credential already has a pending credential intending to replace it.
      */
-    fun createCredential(
+    fun createMdocCredential(
```

Comment:
So, yeah, I think we should probably move credential creation into concrete `Credential`-derived classes and instead of `Document.createCredential()` it should just be `Document.addCredential()` and I think `addCredential()`'s only job would be to save the credential to disk ... 

In this setup the application code would be something like this (for a contrived example where we're adding to mDL credentials - one for LoA Substantial, one for LoA High - one SD/JWT credentiat, and one mDL direct access credential.

```
val mdocCredential = MdocCredential.create(
  domain = "mDL_LoA_Substantial",
  validFrom = validFrom,
  validUntil = validUntil,
  secureArea = androidKeystoreSecureArea,
  createKeySettings = createKeySettings,
  asReplacementFor = null,
)
document.addCredential(mdocCredential)

val mdocCredentialLoAHigh = MdocCredential.create(
  domain = "mDL_LoA_High",
  validFrom = validFrom,
  validUntil = validUntil,
  secureArea = androidKeystoreSecureArea,
  createKeySettings = createKeySettingsForLoAHigh,
  asReplacementFor = null,
)
document.addCredential(mdocCredentialLoAHigh)

val sdjwtCredential = SdJwtCredential.create(
  domain = "sdjwt_LoA_Substantial",
  validFrom = validFrom,
  validUntil = validUntil,
  secureArea = androidKeystoreSecureArea,
  createKeySettings = createKeySettings,
  asReplacementFor = null,
)
document.addCredential(sdjwtCredential)

val mdocDirectAccessCredential = MdocDirectAccessCredential.create(
  domain = "mdoc_directAccess",
  validFrom = validFrom,
  validUntil = validUntil,
  asReplacementFor = null,
  /* some direct-access specific stuff */
)
document.addCredential(mdocDirectAccessCredential)

```

Reply:
since credentials are so tied to the document they're associated with, I added the document to the `create` method so the usage would look like this instead:

```
val mdocCredentialLoAHigh = MdocCredential.create(
  domain = "mDL_LoA_High",
  secureArea = androidKeystoreSecureArea,
  createKeySettings = createKeySettingsForLoAHigh,
  asReplacementFor = null,
  document = document
)
```

with the `document.addCredential` implicitly in the `Credential.create` - lmk what you think